app/handler/transactional_complaint.py

```diff
@@ -0,0 +1,229 @@
+import uuid
+from abc import ABC, abstractmethod
+from io import BytesIO
+from mailbox import Message
+from typing import Optional
+
+from app import s3
+from app.config import (
+    ALERT_COMPLAINT_REPLY_PHASE,
+    ALERT_COMPLAINT_TO_USER,
+    ALERT_COMPLAINT_FORWARD_PHASE,
+)
+from app.email import headers
+from app.email_utils import (
+    get_header_unicode,
+    parse_full_address,
+    save_email_for_debugging,
+    to_bytes,
+    render,
+    send_email_with_rate_control,
+)
+from app.log import LOG
+from app.models import (
+    User,
+    Alias,
+    DeletedAlias,
+    DomainDeletedAlias,
+    Contact,
+    TransactionalComplaint,
+    Phase,
+    TransactionalComplaintState,
+    RefusedEmail,
+)
+
+
+class TransactionalComplaintOrigin(ABC):
+    @classmethod
+    @abstractmethod
+    def get_original_message(cls, message: Message) -> Optional[Message]:
+        pass
+
+    @classmethod
+    @abstractmethod
+    def name(cls):
+        pass
+
+
+class TransactionalYahooOrigin(TransactionalComplaintOrigin):
+    @classmethod
+    def get_original_message(cls, message: Message) -> Optional[Message]:
+        wanted_part = 6
+        for part in message.walk():
+            wanted_part -= 1
+            if wanted_part == 0:
+                return part
+        return None
+
+    @classmethod
+    def name(cls):
+        return "yahoo"
+
+
+class TransactionalHotmailOrigin(TransactionalComplaintOrigin):
+    @classmethod
+    def get_original_message(cls, message: Message) -> Optional[Message]:
```

Comment:
same comment as the one above, I think it'd be helpful to keep the original comment on why we need the part 3.

Reply:
`AutomatedComplaint`? `ProviderComplaint`? I would put some prefix before complaint to distinguish it from a user complaint or some other type of complaint

Reply:
`ProviderComplaint` sounds good.